roboteam_ai/src/conditions/CanShootToTarget.cpp

```diff
@@ -0,0 +1,35 @@
+#include "CanShootToTarget.h"
+
+namespace rtt {
+namespace ai {
+
+CanShootToTarget::CanShootToTarget(std::string name, bt::Blackboard::Ptr blackboard)
+        :Condition(std::move(name), std::move(blackboard)) { };
+
+bt::Node::Status CanShootToTarget::onUpdate() {
+
+    // get the target position to which we want a clear line
+    if (properties->hasVector2("target")) {
+		target = properties->getVector2("target");
+	} else {
+		return bt::Node::Status::Failure;
+	}
+
+    // get the margin, which is the min distance from the line
+	if (properties->hasDouble("margin")) {
+		margin = properties->getDouble("margin");
```

Comment:
in passing, the margin is an angle from the shooting robot in stead of a distance to the passline,
maybe change this to the same definition of can shoot to target